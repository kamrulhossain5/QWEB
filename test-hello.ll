; ModuleID = 'QWEB'
source_filename = "QWEB"

@fmt = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.1 = private unnamed_addr constant [4 x i8] c"%g\0A\00", align 1
@fmt.2 = private unnamed_addr constant [4 x i8] c"%s\0A\00", align 1
@string = private unnamed_addr constant [12 x i8] c"Hello World\00", align 1

declare i32 @prints(i8*, ...)

declare i32 @printbig(i32)

define i32 @main() {
entry:
  %prints = call i32 (i8*, ...) @prints(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt.2, i32 0, i32 0), i8* getelementptr inbounds ([12 x i8], [12 x i8]* @string, i32 0, i32 0))
  ret i32 0
}
