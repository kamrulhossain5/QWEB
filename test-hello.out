FAILED
  ./test-hello.exe failed on ./test-hello.exe
