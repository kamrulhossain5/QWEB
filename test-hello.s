	.section	__TEXT,__text,regular,pure_instructions
	.build_version macos, 11, 0
	.globl	_main                           ## -- Begin function main
	.p2align	4, 0x90
_main:                                  ## @main
	.cfi_startproc
## %bb.0:                               ## %entry
	pushq	%rax
	.cfi_def_cfa_offset 16
	leaq	L_fmt.2(%rip), %rdi
	leaq	L_string(%rip), %rsi
	xorl	%eax, %eax
	callq	_prints
	xorl	%eax, %eax
	popq	%rcx
	retq
	.cfi_endproc
                                        ## -- End function
	.section	__TEXT,__cstring,cstring_literals
L_fmt:                                  ## @fmt
	.asciz	"%d\n"

L_fmt.1:                                ## @fmt.1
	.asciz	"%g\n"

L_fmt.2:                                ## @fmt.2
	.asciz	"%s\n"

L_string:                               ## @string
	.asciz	"Hello World"

.subsections_via_symbols
