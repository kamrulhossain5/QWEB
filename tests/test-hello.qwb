function none main(){
  createHeader("Hello Kam!");
  createSubheader("Hello Kam!");
  createParagraph("Hello Kam!");
  createList("Hello Kam!");
}
