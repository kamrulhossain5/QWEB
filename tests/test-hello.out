<h1>Hello Kam!</h1>
<h2>Hello Kam!</h2>
<p>Hello Kam!</p>
<li>Hello Kam!</li>
